lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPatterns.cpp

```diff
@@ -186,6 +234,13 @@ void addTransposeOpsConversionPatterns(MLIRContext *ctx,
   patterns.add<StableHLOToTTIRTransposeOpConversionPattern>(typeConverter, ctx);
 }
 
+void dotGeneralOpsConversionPatterns(MLIRContext *ctx,
```

Comment:
I guess you didn't like my suggestion to rename to `addMatmulOpsConversionPatterns` 😅 I won't push you but I just wanted to elaborate more about my thinking here. My thinking was that we should group the patterns insertion into some logical groups depending on the resulting class of ops. For example element-wise ops, data-movement ops, matmul ops, etc. If the function is named after specific StableHLO function like `addDotGeneralOpsConversionPatterns` then it can contain only addition of that one pattern, so that almost defeats the purpose of even having that function.

Reply:
Thanks for elaborating your point of view, I don't a strong preference for the patterns function name, just that I think it should be closer to `stablehlo` names so it is easier to identify the op itself from the pattern function name. 

In this case `stablehlo` doesn't have `matmul` op, but rather it has the `stablehlo.dot` and `stablehlo.dot_general` op, from which `dot_general` op is the broader op, so thought to keep it to `addDotOpsConversionPatterns`. But if you have a strong preference to naming it `addMatmulOpsConversionPatterns` then I can change it.

Reply:
I think it fundamentally boils down to, are we naming these functions based on the operator names that we're lowering _from_ or the names that we're lowering _to_?  I'm somewhat indifferent either way, but whatever we pick we should be consistent.

Reply:
> I think it fundamentally boils down to, are we naming these functions based on the operator names that we're lowering from or the names that we're lowering to?

I wanted to make it more universal, to group the ops that lower in the same way (using same OpConversionPattern) and which belong to the same general "class" of ops (eltwise, binary, reduce, matmul, data movement, etc). One goal was obviously to split the code into organized functions, and another was so that if someone is working on ONNX->TTIR conversion and wants to see how for example matmul op is converted for StableHLO, they can easily locate it by finding addMatmulOps function instead of knowing how exactly it is called in StableHLO. I think that matmul is much more recognized name in ML world than dot_general, which is why I wanted to avoid using specific names from any dialect but to stick to some more universal names. Also if we use specific names from some dialect then we condemn these functions to contain only one or two ops, which would explode the number of functions. In any case, I wouldn't name these functions based on the operator names that we're lowering _from_ because sometimes we are lowering couple of ops into one target op, and there it is natural to call the function based on targeting op.

Now I made the whole philosophy about totally unimportant topic, I apologize for that, I have OCD about these little things 😅

Reply:
I agree, I think Usman changed it to read matmul so let's move forward with that.